### PR TITLE
[otelcol] Fix grpclogger to capture correct caller location

### LINKED ISSUE #10772: Clean up gRPC logging
Hello,While working on an  issue of  Jaeger to clean up grpc logs which is :-
https://github.com/jaegertracing/jaeger/issues/5784
The logs are all tagged with zapgrpc/zapgrpc.go:176. This indicates that the Skip() parameter is incorrect when a gRPC logger is
instantiated.
```
...
2024-07-28T16:17:24.271-0400	info	zapgrpc/zapgrpc.go:176	[core] original dial target is: "localhost:4317"	{"grpc_log": true}
2024-07-28T16:17:24.271-0400	info	zapgrpc/zapgrpc.go:176	[core] [Channel #1]Channel created	{"grpc_log": true}
2024-07-28T16:17:24.271-0400	info	zapgrpc/zapgrpc.go:176	[core] [Channel #1]parsed dial target is: resolver.Target{URL:url.URL{Scheme:"dns
```
on furthur digging it seems to be an issue in the grpclog logger i.e. otelcol>internal>grpclog>logger.go which seems to be missing an AddSkipCaller() which is causing logs to be tagged with zapgrpc/zapgrpc.go:176 instead of the real location of the log statements.
**Steps to reproduce**

**What did you expect to see?**
 Logs printing actual locations of the log statements 

**What did you see instead?**
All logs of grpclogger printing zapgrpc/zapgrpc.go:176 

**What version did you use?**
 Version: 0.106.0 

**What config did you use?**

**Environment**
MacOS 14.2.1

**Additional context**

### PR DESCRIPTION
**Pull request by vvs-personalstash**
#### Description
Cleaned up Grpc logs to make them reference the location where the log statements are written instead of zapgrpc/zapgrpc.go:176

#### Link to tracking issue
Fixes #10772
Fixes https://github.com/jaegertracing/jaeger/issues/5784

### CODE REVIEW

File: otelcol/internal/grpclog/logger.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func SetLogger(baseLogger *zap.Logger, loglevel zapcore.Level) *zapgrpc.Logger {
 			c = core
 		}
 		return c.With([]zapcore.Field{zap.Bool("grpc_log", true)})
-	})))
+	}), zap.AddCallerSkip(5)))
```
Comment (yurishkuro) [inline otelcol/internal/grpclog/logger.go:30]: you can probably add a test for this. There is already a unit test that validates some other aspects, you can add a check that the caller is `logger_test.go`

**Comment by yurishkuro** [comment]
Please give a proper title to the PR.

File: otelcol/internal/grpclog/logger.go (hunk 1 above)
Comment (vvs-personalstash) [reply otelcol/internal/grpclog/logger.go:30]: Okay i will work on it.

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
@open-telemetry/collector-maintainers can this be merged?

**Review by codeboten — APPROVED** [review]
Thanks @vvs-personalstash

**Comment by yurishkuro** [comment]
Thanks!

**Pull request merged**